include/swift/AST/Decl.h

```diff
@@ -3536,21 +3532,30 @@ enum class ArtificialMainKind : uint8_t {
   UIApplicationMain,
 };
 
-enum class ObjCClassKind : uint8_t {
-  /// Neither the class nor any of its superclasses are @objc.
-  NonObjC,
-  /// One of the superclasses is @objc but another superclass or the
-  /// class itself has generic parameters, so while it cannot be
-  /// directly represented in Objective-C, it has implicitly @objc
-  /// members.
-  ObjCMembers,
-  /// The top-level ancestor of this class is not @objc, but the
-  /// class itself is.
-  ObjCWithSwiftRoot,
-  /// The class is bona-fide @objc.
-  ObjC,
+enum class AncestryFlags : uint8_t {
+  /// The class or one of its superclasses is @objc.
+  HasObjC = (1<<0),
+
+  /// The class or one of its superclasses is @objcMembers.
+  HasObjCMembers = (1<<1),
+
+  /// The class or one of its superclasses is generic.
+  HasGeneric = (1<<2),
+
+  /// The class or one of its superclasses is resilient.
+  HasResilient = (1<<3),
+
+  /// The class or one of its superclasses has resilient metadata and is in a
+  /// different resilience domain.
+  HasResilientOther = (1<<4),
+
+  /// The class or one of its superclasses is imported.
```

Comment:
What does "imported" mean? Imported from Objective-C? Imported from another module (of any language)?

…can you rename it to HasClangImported? (or just ClangImported, not sure how much the "Has" is buying you on these.)